### PR TITLE
107-zoom

### LINKED ISSUE #107: Port spatial zoom transform
a subtask of #62, port a spatial zoom transform into Monai

ref:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.zoom.html

### PR DESCRIPTION
**Pull request by madil90**
Fixes #107  .

### Description
Add spatial zoom transform

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/transforms.py [hunk 1]
```diff
@@ -80,6 +81,65 @@ def __call__(self, img):
         return np.flip(img, self.axis)
 
 
+@export
+class Zoom:
+    """ Zooms a 3d image. Uses scipy.ndimage.zoom or cupyx.scipy.ndimage.zoom in case of gpu.
```
Comment (Nic-Ma) [inline monai/transforms/transforms.py]: This transform can only be applied to 3D image?
Thanks.

File: monai/transforms/transforms.py [hunk 2]
```diff
@@ -80,6 +81,65 @@ def __call__(self, img):
...
+        if self.use_gpu:
+            try:
+                import cupy
+                from cupyx.scipy.ndimage import zoom as zoom_gpu
+
+                zoomed_gpu = zoom_gpu(cupy.array(img), zoom=self.zoom, order=self.order,
+                                      mode=self.mode, cval=self.cval, prefilter=self.prefilter)
+                zoomed = cupy.asnumpy()
+            except Exception:
+                print('Warning: Zoom gpu failed. Defaulting to cpu.')
```
Comment (wyli) [inline monai/transforms/transforms.py]: i think here we choose gpu or cpu in the constructor, then we don't try and except in every call

Comment (tvercaut) [reply monai/transforms/transforms.py]: Is cupy an approved dependency for MONAI?

File: monai/transforms/transforms.py (hunk 1 above)
Comment (madil90) [reply monai/transforms/transforms.py]: No. that's a mistake. I'll correct it.

File: monai/transforms/transforms.py (hunk 2 above)
Comment (madil90) [reply monai/transforms/transforms.py]: @ericspod @wyli what do you think about cupy? I can totally remove it and leave it as "to-do" for later.

Comment (wyli) [reply monai/transforms/transforms.py]: i feel we don't put cupy in `requirements.txt`,  but leave the cupy calls here. if the user set `use_gpu=True` and this class couldn't import cupy then it raise an error...

Comment (madil90) [reply monai/transforms/transforms.py]: Interesting. I can change ValueError to say "Please install cupy if you want to use gpu zoom".

**Pull request merged**